cql3/statements/schema_altering_statement.cc

```diff
@@ -70,13 +71,15 @@ schema_altering_statement::execute(query_processor& qp, service::query_state& st
             throw std::logic_error(format("Attempted to modify {} via internal query: such schema changes are not propagated and thus illegal", info));
         }
     }
-    auto result = co_await qp.execute_schema_statement(*this, state, options, std::move(guard));
+    service::mutations_collector mc{std::move(guard)};
+    auto result = co_await qp.execute_schema_statement(*this, state, options, mc);
     // We don't want to grant the permissions to the supposed creator even if the statement succeeded if it's an internal query
     // or if the query did not actually create the item, i.e. the query is bounced to another shard or it's a IF NOT EXISTS
     // query where the item already exists.
     if (!internal && result->is_schema_change()) {
-        co_await grant_permissions_to_creator(state.get_client_state());
+        co_await grant_permissions_to_creator(state.get_client_state(), mc);
     }
+    co_await qp.announce_schema_statement(*this, mc);
```

Comment:
But here you announce both schema changes and auth mutations through `schema_change` raft command which sends all mutation to schema merging code.

Reply:
yes, any suggestions?

Reply:
@gleb-cloudius are we good with resolving this conversation or you have further comments?

Reply:
We discussed this on a call some time ago. A rework for raft commands is needed. We should be able to send commands that include mutations for arbitrary combination of tables and trigger updates for relevant subsystems (multiple per command). It should also replace the `mixed_command` that you introduced in your "change rf" series. On that call @piodul also mentioned that auth mutations in the future  will need to trigger some work for service level subsystem so here it will not be possible since auth mutation will be sent inside a schema change command.

On that call we decided that someone need to collect the requirements and do the design. Not sure what happens after that.

Reply:
@gleb-cloudius OK, thanks for the background. I assume it's going to be a longer project and this PR doesn't have to be blocked until it's completed.

Reply:
I do not know what happen after the call. Was there a follow up @piodul @kbr-scylla that we talked about? On one hand I understand that this is small part for this PR and it kind of works (not by design, but by chance), but on the other if everyone just keep pushing staff on top of broken design nobody will work on a fix.

Reply:
@kostja promised us today that this will be planned in @ptrsmrn's team for Q3.
@kostja please post a link to the issue

Reply:
Here's the Issue - https://github.com/scylladb/scylladb/issues/19153
I'll resolve this conversation.